google_guest_agent/uefi/uefi.go

```diff
@@ -0,0 +1,57 @@
+// Package uefi provides utility functions to read UEFI variables.
+package uefi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	defaultEFIVarsDir = "/sys/firmware/efi/efivars"
+)
+
+// VariableName represents UEFI variable name and GUID.
+// Format: {VariableName}-{VendorGUID}
+type VariableName struct {
+	Name string
+	GUID string
+}
+
+// Variable represents UEFI Variable and its contents.
+type Variable struct {
+	Name       VariableName
+	Attributes []byte
+	Content    []byte
+}
+
+// VariablePath returns a path for UEFI variable on disk.
+func VariablePath(v VariableName) string {
+	if v.Name == "testname" && v.GUID == "testguid" {
+		return filepath.Join(os.TempDir(), v.Name+"-"+v.GUID)
+	}
+
+	return filepath.Join(defaultEFIVarsDir, v.Name+"-"+v.GUID)
+}
+
+// ReadVariable reads UEFI variable and returns as byte array.
+// Throws an error if variable is invalid or empty.
+func ReadVariable(v VariableName) (*Variable, error) {
+	path := VariablePath(v)
+	b, err := os.ReadFile(path)
+
+	if err != nil {
+		return nil, fmt.Errorf("error reading %q: %v", path, err)
+	}
+
+	// According to UEFI specification the first four bytes of the contents are attributes.
```

Comment:
This comment makes me wonder if for the mtls context we are using this four bytes to convey anything that's important for us (will change any behavior etc).

Reply:
These attributes are unrelated to content they are basically describing things like if variable is non-volatile/accessible at boot/runtime that won't change the actual content/behavior that we're interested in.